### PR TITLE
[ASM] EXPANDR-6789 - Rcslayout changes

### PR DESCRIPTION
**Pull request by capanw**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/EXPANDR-6789

## Description
A button on my alert page in XSIAM to scan my current asset tied to an alert

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    elif "RCSScanId" in context_get.keys() and scan_id != context_get["RCSScanId"]:
+        # Delete Old Scan ID
+        delete_args = {"key": "RCSScanId"}
+        demisto.executeCommand("DeleteContext", delete_args)
+
+        args_set_scan = {"key": "RCSScanId", "value": scan_id}
+        demisto.executeCommand("Set", args_set_scan)
+        return "Updated RCSScanId Key Value"
+    else:
+        return "RCSScanId remains unchanged"
```
Comment (maimorag) [inline Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py:37]:
```suggestion
    return "RCSScanId remains unchanged"
```

**Comment by capanw** [comment]
@johnnywilkes @BigEasyJ, Can you guys review this. 

As discussed during our demo, made the change to update failure text and also updated code to include @maimorag suggestions. 

Thanks

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    Args:
+        args: A dictionary of arguments passed to the function.
+
+    Returns:
+        A dictionary containing the scan ID, creation status, and scan status.
+    """
+    service_id = args.get("service_id")
+    attack_surface_rule_id = args.get("attack_surface_rule_id")
+    alert_internal_id = args.get("alert_internal_id")
```
[the remark this answers is not in the document]
Comment (capanw) [reply Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py:51]: Thank you for your suggestion. I updated the code and moved this to main funciton.

[the remark this answers is not in the document]
Comment (maimorag) [reply Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py]: @capanw I haven't seen the change, was it committed?

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    output_scan_start = demisto.executeCommand(
+        "asm-start-remediation-confirmation-scan", args_scan_start
+    )
+
+    # Raise error if the command execution failed
+    if output_scan_start[0].get("Type") and "Failed to execute" in output_scan_start[
+        0
+    ].get("Contents"):
+        return_error("Failed to execute RCSScanStatus. Check input values.")
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py:63]: Nit: We talked about making this `raise ValueError` so only one return_error for code

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def main():
+    args = demisto.args()
+    service_id = args.get("service_id")
+    attack_surface_rule_id = args.get("attack_surface_rule_id")
+    alert_internal_id = args.get("alert_internal_id")
+
+    if not all([service_id, attack_surface_rule_id, alert_internal_id]):
+        raise ValueError("All the required arguments needs to be specified")
```
[the remark this answers is not in the document]
Comment (capanw) [reply Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py]: Done, made changes to the code. Thank you!!

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py (hunk 3 above)
Comment (capanw) [reply Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py]: Made changes to this one too. Thank you for bringing this up.

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    output_scan_start = demisto.executeCommand(
+        "asm-start-remediation-confirmation-scan", args_scan_start
+    )
+
+    # Raise error if the command execution failed
+    if output_scan_start[0].get("Type") and "Failed to execute" in output_scan_start[
+        0
+    ].get("Contents"):
+        return "Failed to execute RCSScanStatus. Check input values."
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py:65]:
```suggestion
        raise ValueError("Failed to execute RCSScanStatus. Check input values.")
```
This is what I originally suggested

**Review by johnnywilkes — CHANGES_REQUESTED** [review]

**Review by johnnywilkes — APPROVED** [review]
thanks for making the changes

File: Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py (hunk 5 above)
Comment (capanw) [reply Packs/CortexAttackSurfaceManagement/Scripts/RCSScan/RCSScan.py]: Updated

**Comment by capanw** [comment]
@maimorag, Approved from our side. If all looks good, can you merge it. Thanks

**Review by maimorag — APPROVED** [review]
Amazing work!

**Pull request merged**